### PR TITLE
Update monitor groups

### LINKED ISSUE #189: Monitors & Groups: Updating frontend so it works with backend
### Describe the feature
Updating frontend to use react query and the new database.

### Proposed Solution
Update the code of the page Monitors and Groups from useEffects to fetch data to react query api.
Update Json management to be in conforms with the new database scheme.  

### Other Information
N/A

### PR DESCRIPTION
**Pull request by pdPinho**
### Issue

Closes #189 

### Reason for this change

Due to some changes in our database the backend required some adjustments whilst the frontend had to match those adjustments.

### Description of changes

- Updated `Monitors` and `Specific Monitor` pages to match BD and backend;
- Updated `GroupBar` to match backend;
- Updated and refactored some frontend code to use `react query`;
- Some miscellaneous fixes.

### Description of how you validated changes

Ran `docker compose up`
And on top of that tested multiple routes and functionalities to guarantee that things worked correctly.
 
### Checklist
- [X] I have tested my changes locally.
- [ ] I have updated the documentation to reflect my changes.
- [ ] I have added/updated unit tests (if applicable).
- [X] My code adheres to the [CONTRIBUTING GUIDE](https://ua-smart-signage-platform.github.io/Documentation-Website/guidelines/)

### Aditional Information
Currently missing some verifications but nothing software-breaking experience type.

### CODE REVIEW

**Comment by pdPinho** [comment]
Backend integration tests are currently failing, however it will be ignored for the time being.

File: backend/src/main/java/pt/ua/deti/uasmartsignage/configuration/SecurityConfig.java
Comment (WildBunnie) [inline backend/src/main/java/pt/ua/deti/uasmartsignage/configuration/SecurityConfig.java:1]: is /api/files not what we use for the media player to download the files?

**Review by WildBunnie — CHANGES_REQUESTED** [review]
apart from my question about the api permissions for file downloads all seems ok in the backend

File: backend/src/main/java/pt/ua/deti/uasmartsignage/configuration/SecurityConfig.java
Comment (pdPinho) [reply backend/src/main/java/pt/ua/deti/uasmartsignage/configuration/SecurityConfig.java:1]: You are correct. I will fix it immediately. :+1:

**Pull request merged**